Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,180 @@ Finally, this part describes the OAuth security applied to the API. This spec is
 - Endpoint token URL
 - URL to endpoint authorization ( If flow is based on "`authorizationCode`").	
 
+
+## 12. Subscription, Notification & Event
+
+In order to provided event-based interaction, CAMARA API could provided capabilities for subscription & notification management.
+A subscription allows to an API consumer to request event notification reception at an given url (callback address) for a specific context.
+A notification is the publication at the listener address about an occurred event.
+Event type managed are explicitly defined in CAMAPA API OAS.
+
+### 12.1 Subscription
+
+We distinguish 2 types of subscription:
+- Instance-based subscription (indirect creation)
+- Resource-based subscription (direct creation)
+
+**Instance-based subscription**
+
+An instance-based subscription is a subscription indirectly created, additionnally, to another resource creation. For example for a Payment request (in Carrier Billing API), in the `POST/payments`, the API consumer could request to get event notification about **this** Payment request processing update. The subscription is not an entity and its lifecycle is linked to the managed entity (the Payment resource in this case). The subscription terminates with the managed entity.
+
+Providing this capability is optionnal for any CAMARA APIs depending on UC requirements
+
+If this capability is present in CAMARA API, following attributes **must** be used in the POST request for the managed entity:
+
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| notificationUrl | string |callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentification token for callback API | optional |
+
+
+**Resource-based subscription**
+
+A resource-based subscription is is subscription managed as a resource. A dedicated endpoint is provided to request subscription creation.  As this subscription is managed API resource, it is identified and operations to search, retrieve and delete them must be provided.
+
+Note: It is perfectly valid for a CAMARA API to have several subscriptions endpoints when distinct events are managed (for example in Device status API we manage distinct subscription for roaming status and reachability status event)
+
+In order to ease developer adoption, the pattern for Resource-based subscription must be the consistent for all API providing this feature.
+
+4 operations must be defined:
+
+| operation | path | description |
+| ----- |	-----  |	 -----  | 
+| POST | `/subscriptions/<subscriptionName>` | Operation to request a subscription.  eg. POST /subscriptions/roamingStatus    |
```

Comment:
By means of POST we are creating a new subscription. It confuses to indicate susbcriptionName within the path. 
It makes sense to have in the body.

Reply:
Ok! seems you're aligned with @akoshunyadi perspective about having a generic subscription path and not having the subscription name in the path. I'm fine with this pattern.